temporal-sdk/src/main/java/io/temporal/client/WorkflowStub.java

```diff
@@ -73,6 +74,92 @@ static <T> WorkflowStub fromTyped(T typed) {
    */
   void signal(String signalName, Object... args);
 
+  /**
+   * Synchronously update a workflow execution by invoking its update handler. Usually a update
+   * handler is a method annotated with {@link io.temporal.workflow.UpdateMethod}.
+   *
+   * @param updateName name of the update handler. Usually it is a method name.
+   * @param resultClass class of the update return value
+   * @param <R> type of the update return value
+   * @param args update method arguments
+   * @return update result
+   * @throws WorkflowNotFoundException if the workflow execution doesn't exist or completed and
+   *     can't be signalled
+   * @throws WorkflowServiceException for all other failures including networking and service
+   *     availability issues
+   */
+  @Experimental
+  <R> R update(String updateName, Class<R> resultClass, Object... args);
+
+  /**
+   * Synchronously update a workflow execution by invoking its update handler. Usually a update
+   * handler is a method annotated with {@link io.temporal.workflow.UpdateMethod}.
+   *
+   * @param updateName name of the update handler. Usually it is a method name.
+   * @param updateId is an application-layer identifier for the requested update. It must be unique
+   *     within the scope of a workflow execution.
+   * @param firstExecutionRunId specifies the RunID expected to identify the first run in the
+   *     workflow execution chain. If this expectation does not match then the server will reject
+   *     the update request with an error.
+   * @param resultClass class of the update return value.
+   * @param <R> type of the update return value.
+   * @param resultType type of the workflow return value. Differs from resultClass for generic
+   *     types.
+   * @param args update method arguments
+   * @return update result
+   * @throws WorkflowNotFoundException if the workflow execution doesn't exist or completed and
+   *     can't be sent an update request
+   * @throws WorkflowServiceException for all other failures including networking and service
+   *     availability issues
+   */
+  @Experimental
+  <R> R update(
+      String updateName,
+      String updateId,
+      String firstExecutionRunId,
+      Class<R> resultClass,
+      Type resultType,
+      Object... args);
+
+  /**
+   * Asynchronously update a workflow execution by invoking its update handler and returning a
+   * reference to the update request. Usually a update handler is a method annotated with {@link
+   * io.temporal.workflow.UpdateMethod}.
+   *
+   * @param updateName name of the update handler. Usually it is a method name.
+   * @param resultClass class of the update return value
+   * @param <R> type of the update return value
+   * @param args update method arguments
+   * @return update reference that can be used to get the result of the update.
+   */
+  @Experimental
+  <R> UpdateReference<R> updateAsync(String updateName, Class<R> resultClass, Object... args);
```

Comment:
> Or maybe the utility is so dumb and easy that it's not necessary

That's true in Go, but Java with it's overloads and default interface impls it may be worth it. In other languages we often have "start" + "handle get", and then that combination we make a shortcut for, but not Go.